commit 7c8ffc946066bc7d572be2648490953f6ccd30d6
Author: John Johansen <john.johansen@canonical.com>
Date:   Thu Jul 21 11:12:38 2016 -0700

    UBUNTU: SAUCE: apparmor: fix stack trace when removing namespace with profiles
    
    BugLink: http://bugs.launchpad.net/bugs/1593874
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>

diff --git a/security/apparmor/policy_ns.c b/security/apparmor/policy_ns.c
index d06e664..19adb24 100644
--- a/security/apparmor/policy_ns.c
+++ b/security/apparmor/policy_ns.c
@@ -259,9 +259,13 @@ static void destroy_ns(struct aa_ns *ns)
 	/* release all sub namespaces */
 	__ns_list_release(&ns->sub_ns);
 
-	if (ns->parent)
+	if (ns->parent) {
+		unsigned long flags;
+		write_lock_irqsave(&ns->labels.lock, flags);
 		__aa_proxy_redirect(ns_unconfined(ns),
 				    ns_unconfined(ns->parent));
+		write_unlock_irqrestore(&ns->labels.lock, flags);
+	}
 	__aa_fs_ns_rmdir(ns);
 	mutex_unlock(&ns->lock);
 }
